### PR TITLE
Fix bugs

### PR DESCRIPTION
**Pull request by martinpopel**
I've added `SpaceAfter=No` and fixed several other errors
(described in individual commits of this PR).

@olesar: you can check the commits one by one.
Alternatively, you can see all the changes with:
```
wget http://ufallab.ms.mff.cuni.cz/~popel/lt/{old,new}.txt
vimdiff {old,new}.txt
```

[bugs.html](http://ufallab.ms.mff.cuni.cz/~popel/lt/bugs.html) shows few remaining problems
(which can be fixed in UDv2.1, not now).

### CODE REVIEW

**Review by olesar — APPROVED** [review]

**Review by olesar — COMMENTED** [review]
NOUN to PROPN change: not sure what is the rule for the names of songs, movies, pieces, etc. If it is just one word then we change it to PROPN, and if not... (e.g. "Three men in a boat")?

**Review by olesar — COMMENTED** [review]
Degree=Pos removed: This is not documented in the guidelines. Degree=Pos is a universal feature
http://universaldependencies.org/u/feat/all.html#Pos

File: lt-ud-train.conllu [hunk 1]
```diff
@@ -94,7 +94,7 @@
 8	drąsų	drąsųs	ADJ	JJL	Case=Acc|Definite=Ind|Degree=Pos|Gender=Masc|Number=Sing	9	amod	_	En=brave
 9	žingsnį	žingsnis	NOUN	NN	Case=Acc|Gender=Masc|Number=Sing	7	obj	_	En=step
 10	ir	ir	CCONJ	CC	_	12	cc	_	En=and
-11	viešai	viešai	ADV	RB	Degree=Pos	13	ccomp	_	En=publicly
+11	viešai	viešai	ADV	RB	Degree=Pos	17	ccomp	_	En=publicly
```
Comment (olesar) [inline lt-ud-train.conllu]: HeadPos: 12

File: lt-ud-train.conllu [hunk 2]
```diff
@@ -224,7 +224,7 @@
 45	rašytojams	rašytojas	NOUN	NN	Case=Dat|Gender=Masc|Number=Plur	7	conj	_	En=writer
 46	Daivai	daiva	PROPN	NNP	Case=Dat|Gender=Fem|Number=Sing	45	flat	_	En=Daiva
 47	Čepauskaitei	čepauskaitė	PROPN	NNP	Case=Dat|Gender=Fem|Number=Sing	46	flat	_	En=Čepauskaitė
-48	ir	ir	CCONJ	CC	_	37	cc	_	En=and
+48	ir	ir	CCONJ	CC	_	38	cc	_	En=and
```
Comment (olesar) [inline lt-ud-train.conllu]: HeadPos: 49

File: lt-ud-train.conllu [hunk 3]
```diff
@@ -1115,7 +1115,7 @@
 # sent_id = 36
 # text = Liko Aristofanas, nes jis gerai rašė (kaip pasakytų Audenas, time pardoned him for writing well).
 1	Liko	likti	VERB	VBC	Mood=Ind|Number=Sing|Person=3|Polarity=Pos|Reflex=No|Tense=Past|VerbForm=Fin|Voice=Act	0	root	_	En=remain
-2	Aristofanas	Aristofanas	PROPN	NNP	Case=Nom|Gender=Masc|Number=Sing	3	nsubj	_	En=Aristofane|SpaceAfter=No
+2	Aristofanas	Aristofanas	PROPN	NNP	Case=Nom|Gender=Masc|Number=Sing	7	nsubj	_	En=Aristofane|SpaceAfter=No
```
Comment (olesar) [inline lt-ud-train.conllu]: HeadPos: 1

File: lt-ud-train.conllu [hunk 4]
```diff
@@ -1131,7 +1131,7 @@
 15	him	him	X	X	_	16	flat	_	En=_
 16	for	for	X	X	_	17	flat	_	En=_
 17	writing	writing	X	X	_	18	flat	_	En=_
-18	well	well	X	X	_	19	flat	_	En=_|SpaceAfter=No
+18	well	well	X	X	_	13	flat	_	En=_|SpaceAfter=No
```
Comment (olesar) [inline lt-ud-train.conllu]: Nr 14-18 should be 13?

File: lt-ud-train.conllu [hunk 5]
```diff
@@ -1149,12 +1149,12 @@
 # newpar id = 38
 # sent_id = 38
 # text = Tiesa, Strepsiadas ir negalėjo laimėti.
-1	Tiesa	Tiesa	NOUN	NN	Case=Nom|Gender=Fem|Number=Sing	2	parataxis	_	En=to_say_the_truth|SpaceAfter=No
+1	Tiesa	Tiesa	NOUN	NN	Case=Nom|Gender=Fem|Number=Sing	3	parataxis	_	En=to_say_the_truth|SpaceAfter=No
```
Comment (olesar) [inline lt-ud-train.conllu]: HeadPos: 5

File: lt-ud-train.conllu [hunk 6]
```diff
@@ -1313,7 +1313,7 @@
 15	keliu	kelias	NOUN	NN	Case=Ins|Gender=Masc|Number=Sing	9	iobj	_	En=path|SpaceAfter=No
 16	,	,	PUNCT	PUNCT	_	30	punct	_	En=,
 17	nors	nors	SCONJ	UH	_	30	mark	_	En=though
-18	jau	jau	PART	RB	Degree=Pos	24	advmod	_	En=already
+18	jau	jau	PART	RB	Degree=Pos	28	advmod	_	En=already
```
Comment (olesar) [inline lt-ud-train.conllu]: HeadPos: 23

File: lt-ud-train.conllu [hunk 7]
```diff
@@ -1714,7 +1714,7 @@
 6	ne	ne	PART	UH	Polarity=Neg	7	advmod	_	En=not
 7	lietuvių	lietuvis	NOUN	NN	Case=Gen|Definite=Ind|Degree=Pos|Gender=Masc|Number=Plur	8	nmod	_	En=Lithianian
 8	spaudoje	spauda	NOUN	NN	Case=Loc|Gender=Fem|Number=Sing	2	obl	_	En=press|SpaceAfter=No
-9	,	,	PUNCT	PUNCT	_	10	punct	_	En=,
+9	,	,	PUNCT	PUNCT	_	11	punct	_	En=,
```
Comment (olesar) [inline lt-ud-train.conllu]: HeadPos: 8

File: lt-ud-train.conllu [hunk 8]
```diff
@@ -1786,7 +1786,7 @@
 5	pas	pas	ADP	UH	_	6	case	_	En=to
 6	jus	tu	PRON	PRP	Case=Acc|Number=Plur|Person=2	7	obl	_	En=you
 7	vadinami	vadinti	VERB	VBC	Case=Nom|Definite=Ind|Gender=Masc|Number=Plur|Polarity=Pos|Reflex=No|Tense=Pres|VerbForm=Part|Voice=Act	8	acl:relcl	_	En=call
-8	dešiniaisiais	dešinys	ADJ	JJL	Case=Ins|Definite=Def|Degree=Pos|Gender=Masc|Number=Plur	9	obj	_	En=right-wing|SpaceAfter=No
+8	dešiniaisiais	dešinys	ADJ	JJL	Case=Ins|Definite=Def|Degree=Pos|Gender=Masc|Number=Plur	10	obj	_	En=right-wing|SpaceAfter=No
```
Comment (olesar) [inline lt-ud-train.conllu]: HeadPos: 7

File: lt-ud-train.conllu [hunk 9]
```diff
@@ -2042,7 +2042,7 @@
 29	ir	ir	CCONJ	CC	_	30	cc	_	En=and
 30	galėsi	galėti	VERB	VBC	Mood=Ind|Number=Sing|Person=2|Polarity=Pos|Reflex=No|Tense=Fut|VerbForm=Fin|Voice=Act	28	conj	_	En=can
 31	daryti	daryti	VERB	VB	Polarity=Pos|Reflex=No|VerbForm=Inf|Voice=Act	30	xcomp	_	En=make
-32	karjerą	karjera	NOUN	NN	Case=Acc|Gender=Fem|Number=Sing	33	obj	_	En=carrier|SpaceAfter=No
+32	karjerą	karjera	NOUN	NN	Case=Acc|Gender=Fem|Number=Sing	34	obj	_	En=carrier|SpaceAfter=No
```
Comment (olesar) [inline lt-ud-train.conllu]: HeadPos: 31

File: lt-ud-train.conllu [hunk 10]
```diff
@@ -2285,7 +2285,7 @@
 5	nacionalizmas	nacionalizmas	NOUN	NN	Case=Nom|Gender=Masc|Number=Sing	1	nsubj	_	En=nationalism
 6	–	–	PUNCT	PUNCT	_	7	punct	_	En=–
 7	sakyčiau	sakyti	VERB	VBNL	Mood=Cnd|Number=Sing|Person=1|Polarity=Pos|Reflex=No|VerbForm=Fin|Voice=Act	10	parataxis	_	En=say|SpaceAfter=No
-8	,	,	PUNCT	PUNCT	_	9	punct	_	En=,
+8	,	,	PUNCT	PUNCT	_	10	punct	_	En=,
```
Comment (olesar) [inline lt-ud-train.conllu]: HeadPos: 7

File: lt-ud-train.conllu [hunk 11]
```diff
@@ -2351,7 +2351,7 @@
 20	atsiradusi	atsirasti	VERB	VBNL	Case=Nom|Definite=Ind|Gender=Fem|Number=Sing|Polarity=Pos|Reflex=Yes|Tense=Past|VerbForm=Part|Voice=Act	23	amod	_	En=arise
 21	vartotojiškų	vartotojiškas	ADJ	JJL	Case=Gen|Definite=Ind|Degree=Pos|Gender=Masc|Number=Plur	23	amod	_	En=consumer
 22	siekių	siekis	NOUN	NN	Case=Gen|Gender=Masc|Number=Plur	23	nmod	_	En=will_to_succeed
-23	frustracija	frustracija	NOUN	NN	Case=Nom|Gender=Fem|Number=Sing	24	conj	_	En=frustration
+23	frustracija	frustracija	NOUN	NN	Case=Nom|Gender=Fem|Number=Sing	31	conj	_	En=frustration
```
Comment (olesar) [inline lt-ud-train.conllu]: HeadPos: 16

File: lt-ud-train.conllu [hunk 12]
```diff
@@ -2431,7 +2431,7 @@
 5	šioje	šis	DET	DT	Case=Loc|Definite=Ind|Gender=Fem|Number=Sing	6	det	_	En=this
 6	bėdoje	bėda	NOUN	NN	Case=Loc|Gender=Fem|Number=Sing	8	obl	_	En=adversity
 7	nesame	būti	AUX	VBC	Mood=Ind|Number=Plur|Person=1|Polarity=Neg|Reflex=No|Tense=Pres|VerbForm=Fin|Voice=Act	8	cop	_	En=be_not
-8	vieniši	vienišas	ADJ	JJL	Case=Nom|Definite=Ind|Degree=Pos|Gender=Masc|Number=Plur	9	ccomp	_	d|En=alone|SpaceAfter=No
+8	vieniši	vienišas	ADJ	JJL	Case=Nom|Definite=Ind|Degree=Pos|Gender=Masc|Number=Plur	1	ccomp	_	d|En=alone|SpaceAfter=No
```
Comment (olesar) [inline lt-ud-train.conllu]: HeadPos: 2

File: lt-ud-train.conllu [hunk 13]
```diff
@@ -2470,7 +2470,7 @@
 31	toli	toli	ADV	RB	Degree=Pos	32	advmod	_	En=far
 32	nukrypusi	nukrypti	VERB	VBC	Case=Nom|Definite=Ind|Gender=Fem|Number=Sing|Polarity=Pos|Reflex=No|Tense=Past|VerbForm=Part|Voice=Act	25	conj	_	En=tend
 33	į	į	ADP	UH	_	34	case	_	En=to
-34	dešinę	dešinė	NOUN	NN	Case=Acc|Gender=Fem|Number=Sing	35	obl	_	En=right|SpaceAfter=No
+34	dešinę	dešinė	NOUN	NN	Case=Acc|Gender=Fem|Number=Sing	25	obl	_	En=right|SpaceAfter=No
```
Comment (olesar) [inline lt-ud-train.conllu]: HeadPos: 32

File: lt-ud-train.conllu [hunk 14]
```diff
@@ -2726,7 +2726,7 @@
 15	net	net	PART	UH	_	16	advmod	_	En=even
 16	negirdėjo	girdėti	VERB	VBC	Mood=Ind|Number=Sing|Person=3|Polarity=Neg|Reflex=No|Tense=Past|VerbForm=Fin|Voice=Act	2	advcl	_	En=hear
 17	to	tas	PRON	DT	Case=Gen|Gender=Masc|Number=Sing	18	fixed	_	En=that
-18	vardo	vardas	NOUN	NN	Case=Gen|Gender=Masc|Number=Sing	19	nmod	_	En=|SpaceAfter=No
+18	vardo	vardas	NOUN	NN	Case=Gen|Gender=Masc|Number=Sing	2	nmod	_	En=|SpaceAfter=No
```
Comment (olesar) [inline lt-ud-train.conllu]: HeadPos: 16

File: lt-ud-train.conllu [hunk 15]
```diff
@@ -2726,7 +2726,7 @@
 15	net	net	PART	UH	_	16	advmod	_	En=even
 16	negirdėjo	girdėti	VERB	VBC	Mood=Ind|Number=Sing|Person=3|Polarity=Neg|Reflex=No|Tense=Past|VerbForm=Fin|Voice=Act	2	advcl	_	En=hear
 17	to	tas	PRON	DT	Case=Gen|Gender=Masc|Number=Sing	18	fixed	_	En=that
```
Comment (olesar) [inline lt-ud-train.conllu]: POS: DET, DEPREL: det

File: lt-ud-train.conllu [hunk 16]
```diff
@@ -3345,7 +3345,7 @@
 3	pat	pat	PART	RB	Degree=Pos	2	fixed	_	En=well
 4	psichologinio	psichologinis	ADJ	JJL	Case=Gen|Definite=Ind|Degree=Pos|Gender=Masc|Number=Sing	5	nmod	_	En=psychologist
 5	komplekso	kompleksas	NOUN	NN	Case=Gen|Gender=Masc|Number=Sing	0	root	_	En=complex|SpaceAfter=No
-6	,	,	PUNCT	PUNCT	_	9	punct	_	En=,
+6	,	,	PUNCT	PUNCT	_	10	punct	_	En=,
```
Comment (olesar) [inline lt-ud-train.conllu]: HeadPos: 8

File: lt-ud-train.conllu [hunk 17]
```diff
@@ -3483,14 +3483,14 @@
 11	įmanoma	įmanomas	ADJ	JJL	Case=Nom|Definite=Ind|Degree=Pos|Gender=Fem|Number=Sing	14	advcl	_	En=|SpaceAfter=No
 12	,	,	PUNCT	PUNCT	_	14	punct	_	En=,
 13	tai	tai	SCONJ	UH	_	14	mark	_	En=this
-14	žalinga	žalingas	ADJ	JJL	Case=Nom|Definite=Ind|Degree=Pos|Gender=Fem|Number=Sing	15	amod	_	En=harmful|SpaceAfter=No
+14	žalinga	žalingas	ADJ	JJL	Case=Nom|Definite=Ind|Degree=Pos|Gender=Fem|Number=Sing	7	amod	_	En=harmful|SpaceAfter=No
```
Comment (olesar) [inline lt-ud-train.conllu]: DEPREL: conj

File: lt-ud-train.conllu [hunk 18]
```diff
@@ -3483,14 +3483,14 @@
...
 15	.	.	PUNCT	PUNCT	_	7	punct	_	En=.
 
 # newpar id = 131
 # sent_id = 131
 # text = Ji yra anachroniškas mentalinis konstruktas, paveldėtas iš XIX amžiaus svajotojų.
 1	Ji	ji	PRON	PRP	Case=Nom|Gender=Fem|Number=Sing|Person=3	5	nsubj	_	En=it
-2	yra	būti	AUX	VBC	Mood=Ind|Number=Sing|Person=3|Polarity=Pos|Reflex=No|Tense=Pres|VerbForm=Fin|Voice=Act	6	cop	_	En=be
+2	yra	būti	AUX	VBC	Mood=Ind|Number=Sing|Person=3|Polarity=Pos|Reflex=No|Tense=Pres|VerbForm=Fin|Voice=Act	7	cop	_	En=be
```
Comment (olesar) [inline lt-ud-train.conllu]: HeadPos: 5

**Review by olesar — APPROVED** [review]

**Review by olesar — APPROVED** [review]

**Review by olesar — APPROVED** [review]

**Review by olesar — APPROVED** [review]

File: lt-ud-train.conllu [hunk 19]
```diff
@@ -2819,7 +2819,7 @@
 # sent_id = 101
 # text = Vytautas Radžvilas kalba apie Lietuvos išvalstybinimą, nutautinimą, valstybinės ir tautinės sąmonės dekonstrukciją Europos Sąjungoje.
 1	Vytautas	Vytautas	PROPN	NNP	Case=Nom|Gender=Masc|Number=Sing	2	nsubj	_	En=Vytautas
-2	Radžvilas	Radžvilas	PROPN	NNP	Case=Nom|Gender=Masc|Number=Sing	3	flat	_	En=Radžvilas
+2	Radžvilas	Radžvilas	PROPN	NNP	Case=Nom|Gender=Masc|Number=Sing	3	appos	_	En=Radžvilas
```
Comment (olesar) [inline lt-ud-train.conllu]: flat!

**Comment by martinpopel** [comment]
@olesar: thanks for the thorough review.
First one general note.  There are three ways (which can be combined) to apply your suggestions (mostly the change of head):
* If there was an error in the original data, you can fix it and push to the repo. I will cancel this pull request, apply my scripts on the new data and create a new pull request.
* If you don't agree with some of the commits (e.g. flat->appos) or if you feel the commit caused more troubles than good, I can reapply my scripts and omit that commit. Or I can try to fine-tune the given rule.
* Some of the changes may be done manually after merging the pull request, but I don't have time for doing the manual annotation myself. I am sorry I have not explained this before.

**Comment by martinpopel** [comment]
> NOUN to PROPN change: If it is just one word then we change it to PROPN, and if not... (e.g. "Three men in a boat")?

The [PROPN specification](http://universaldependencies.org/u/pos/PROPN.html) gives an example:
>> in *Cat on a Hot Tin Roof*, Cat is NOUN, on is ADP, a is DET, etc.

UD does not attempt to do proper Named Entity Recognition.

> what is the rule for the names of songs, movies, pieces, etc.

My heuristics for changing NOUN to PROPN was based on whether the lemma is capitalized.
I admit that e.g. Debesys as a name of a [comedy](https://en.wikipedia.org/wiki/The_Clouds)
may/should be annotated as NOUN and the lemma lowercased.
I could use automatic rule based on the English gloss or this could be done manually.

**Comment by martinpopel** [comment]
> Degree=Pos removed: This is not documented in the guidelines. Degree=Pos is a universal feature

Yes, it it a universal feature, but intended for ADJectives and ADVerbs only.
I've copied this test from [svalidation](http://universaldependencies.org/svalidation.html#degree-feature-used-with-a-word-that-is-neither-adjective-nor-adverb).
If you don't agree with this, I can simply omit this deletion from the PR.
However, I think Degree=Pos does not add any new info.

**Comment by olesar** [comment]
@martinpopel I see, I didn't noticed the condition.23.02.2017, 04:15, "Martin Popel" <notifications@github.com>:
Degree=Pos removed: This is not documented in the guidelines. Degree=Pos is a universal feature

Yes, it it a universal feature, but intended for ADJectives and ADVerbs only.
I've copied this test from svalidation.
If you don't agree with this, I can simply omit this deletion from the PR.
However, I think Degree=Pos does not add any new info.

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




Olga Lyashevskaya School of Linguistics, Faculty of Humanities Higher School of Economics, Moscow

File: lt-ud-train.conllu (hunk 19 above)
Comment (martinpopel) [reply lt-ud-train.conllu]: Both flat and appos should be head initial, so obviously there was an error in the original data and there is another error in this new data.
For this reason it is listed in [bugs.html](http://ufallab.ms.mff.cuni.cz/~popel/lt/bugs.html) as a ToDo for the next release.

I don't understand this sentence, where the main predicate "kalba" is annotated as NOUN (shouldn't it be a VERB as the features suggest?).
If  "Vytautas Radžvilas kalba apie" means "Vytautas Radžvilas speaks about", you are right Radžvilas should have deprel=flat (or flat:name), but it should not depend on kalba, but on Vytautas. Once this is fixed in the original data, my script won't change it to appos anymore.

**Comment by martinpopel** [comment]
Closing this PR.
@olesar pushed a new version of the data to the dev branch (manually fixing some of the errors highlighted in this PR).
I've created a new PR #2 which is applied on top of the new data and which supersedes this PR.

**Pull request closed without merging**